graphix/sim/base_backend.py

```diff
@@ -22,19 +22,19 @@ def perform_measure(
     if pr_calc:
         op_mat = op_mat_from_result(vec, False)
         prob_0 = state.expectation_single(op_mat, qubit)
-        result = rng.rand() > prob_0
+        result = rng.random() > prob_0
         if result:
             op_mat = op_mat_from_result(vec, True)
     else:
         # choose the measurement result randomly
-        result = np.random.choice([0, 1])
+        result = rng.choice([0, 1])
         op_mat = op_mat_from_result(vec, result)
     state.evolve_single(op_mat, qubit)
     return result
 
 
 class Backend:
-    def __init__(self, pr_calc: bool = True):
+    def __init__(self, pr_calc: bool = True, rng: np.random.Generator = None):
```

Comment:
Please use `| None` explicitly as specified in PEP.

Reply:
Fixed in dbd90dc.